### DISCUSSION TITLE
How to apply a saved view?

### DISCUSSION THREAD

**Discussion by ksurell** [iTwin/itwinjs-core#2649] [Q&A] [answered]
Is it possible to apply a Saved View to a viewport?
I've been looking at IModelConnection, SavedView, ViewState but cannot find a way to retrieve a saved view.

It looks as if I can apply a saved view with:
    _const viewState = await SavedView.viewStateFromProps(imodel, viewProp);
    if (viewState) {
      vp.changeView(viewState);
     }_
but I cannot figure out how to get the viewProp. Maybe I'm looking in the wrong place?

/Krister

**Comment by aruniverse** [comment, marked as answer]
Have you taken a look at this [sample](https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=serialize-view-sample&imodel=Metrostation+Sample)?

**Reply by ksurell** [reply]
Nope! Looks like what I need.

/Krister

**Comment by ksurell** [comment]
I discovered that I can get info about the saved view through the query:
_select * from BisCore.ViewDefinition3d where CodeValue like 'Light%'_
I can get the camera params from here. Maybe _Origin_ and _EyePoint_ will do to position the camera?

 Now I just need help on how to apply them to the viewport.

/Krister

**Reply by pmconne** [reply]
@ksurell,

Rewrite your query to select just the ECInstanceId of the view. Pass the Id to `IModelConnection.views.load` which will return a ViewState that you can subsequently pass to `Viewport.changeView`.
```
async function applyView(viewport: Viewport, viewId: Id64String): Promise<void> {
  const view: ViewState = await viewport.iModel.views.load(viewId);
  viewport.changeView(view);
}
```

Many more [details here](https://www.itwinjs.org/learning/frontend/views/#loading-views-from-an-imodel).

**Reply by ksurell** [reply]
Thank, that worked just fine!

/Krister
